tools/testing/tools_test.go

```diff
@@ -0,0 +1,67 @@
+package toolstesting
+
+import (
+	"errors"
+	"sync"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+
+	"github.com/symflower/eval-dev-quality/tools"
+)
+
+func TestRequiresTool(t *testing.T) {
+	type testCase struct {
+		Name string
+
+		SetupMock    func(t *testing.T, m *MockTool)
+		ExpectFailed bool
+	}
+
+	validate := func(t *testing.T, tc *testCase) {
+		t.Run(tc.Name, func(t *testing.T) {
+			tt := &testing.T{}
+			mock := NewMockToolNamed(t, "some-tool")
+			tc.SetupMock(t, mock)
+
+			// The "testing" logic internally calls "runtime.Goexit" in case of a failure which exits a Goroutine (but still runs "defer"s) so start a new one to not kill ourselves.
+			var waitGroup sync.WaitGroup
```

Comment:
Why do we need a waitgroup? This code does not start a goroutine and IMO it shouldn't in general start one. That would be surprising.

Reply:
Explanation is right above. If we have a test case where `RequiresTool` fails and calls `FailNow`, that will internally call `runtime.Goexit`. That would kill "ourselves" (I had that already, that's why I introduced the goroutine). So we need a sub-goroutine that is killed instead.

Reply:
Also... this code **does** start a goroutine... right below :thinking:

Reply:
> Also... this code **does** start a goroutine... right below 🤔

The testing code you mean?

Reply:
> Explanation is right above. If we have a test case where `RequiresTool` fails and calls `tt.FailNow`, that will internally call `runtime.Goexit`. That would kill "ourselves" (I had that already, that's why I introduced the goroutine). So we need a sub-goroutine that is killed instead.

When you have code that does "exit" calls, test from outside. Not in the same process.

Reply:
I really think that is overly complicated... `testing.T.FailNow` calls `runtime.Goexit` but that is an internal technicality that we unfortunately run into here. Is it really worth it making this a system tests that includes writing a file and calling `go test`? I guess alternatively I could mock `testing.T`... a bit less complex.

Reply:
@zimmski plz take another look at this

Reply:
Looks good to me.